Adding 2 variables to disable ctime and atime during backups

NissesSenap opened the pull request:
By default if netbackup backups a file linux updates the ctime and atime
values of the file.
If you for different reasons don't want this to happen you can use
DO_NOT_RESET_FILE_ACCESS_TIME=YES
USE_CTIME_FOR_INCREMENTALS=YES
When nothing is defined in the config file the default is NO.

Phil-Friderici commented:
Hej @NissesSenap,

both params seems to be boolean if set. My suggestion would be to use real booleans for them and convert them to 'yes' or 'no' later on. For an example check the handling of ```clear_at_boot``` in fscleanup [1].

As far as I can see the lines will not be printed when undef. So the default (of the module) is to not include the line, maybe the default of the application is NO if nothing is defined.

Missing tests:
- proof that the lines are not included when undef (in the default test case).
- proof that the lines are included when set (in specific 'atomic' tests)

[1] https://github.com/Phil-Friderici/puppet-module-fscleanup/blob/master/manifests/init.pp

NissesSenap commented:
Hej @Phil-Friderici what do you mean with 
- proof that the lines are included when set (in specific 'atomic' tests)

I see now that i forgot to add spec test for both of the variabels. If you think it looks okay i will add it for the second variable as well.

Phil-Friderici commented on spec/classes/client_spec.rb:
```diff
@@ -493,5 +493,25 @@
+    end
+    context 'when use_ctime_for_incrementals set to valid undef' do
+      let(:params) { { :use_ctime_for_incrementals => 'undef' } }
+      it { should contain_file('bp_config').with_content(%r{^$}) }
```
Good opportunity to be more specific:
`it { should_not contain_file('bp_config').with_content(%r{^USE_CTIME_FOR_INCREMENTALS.*$}) }`

Phil-Friderici commented:
> Hej @Phil-Friderici what do you mean with
>> proof that the lines are included when set (in specific 'atomic' tests)

I meant something like this: https://github.com/NissesSenap/puppet-module-netbackup/blob/4e25072934a34b108f2fe933248424126b0a6c1a/spec/classes/client_spec.rb#L508-L511

> I see now that i forgot to add spec test for both of the variabels. If you think it looks okay i will add it for the second variable as well.

Yes we are on track, please go ahead.

NissesSenap commented:
@Phil-Friderici i noticed this when i sent in the value as undef:
Notice: /Stage[main]/Netbackup::Client/Notify[access_time]/message: defined 'message' as 'access_time: '
Notice: access_time_bool: false
Notice: /Stage[main]/Netbackup::Client/Notify[access_time_bool]/message: defined 'message' as 'access_time_bool: false'
Notice: access_time_str: NO
Notice: /Stage[main]/Netbackup::Client/Notify[access_time_str]/message: defined 'message' as 'access_time_str: NO'

Seems like the str2bool converts a undef to false which kind of make sense and that's why it didn't work in the teamplate when i started to use _string.
Due to this i have reverted that in the template and updated the spec test to match.
I hope this is okay, i know that we won't get the nicest error but we will still get one.

Hope you think it looks okay.

Phil-Friderici commented:
Thanks @NissesSenap, released in v0.6.0 :)
